src/engine/cache/file-cache.js

```diff
@@ -118,10 +118,19 @@ class FileCache extends BaseCache {
     }
 
     const sortedFiles = fileNames
-      .map(async (fileName) => ({
-        path: path.resolve(this.fileFolder, fileName),
-        timestamp: (await fs.stat(path.resolve(this.fileFolder, fileName))).mtime.getTime(),
-      }))
+      .map(async (fileName) => {
+        let fileState = null
+        try {
+          fileState = await fs.stat(path.resolve(this.fileFolder, fileName))
+        } catch (error) {
+          this.logger.error(error)
+        }
+        return {
+          path: path.resolve(this.fileFolder, fileName),
+          timestamp: fileState ? fileState.mtime.getTime() : null,
+        }
+      })
+      .filter((file) => file.timestamp !== null)
```

Comment:
adding comment here would be useful since it is not particularly trivial what is achieved

Reply:
Indeed, on it.